File: src/main/java-graphql/io/github/graphql/model/StarredRepositoryEdgeResponseProjection.java
package io.github.graphql.model;

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField;
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection;

/**
 * Response projection for StarredRepositoryEdge
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-06T19:47:48+0800"
)
public class StarredRepositoryEdgeResponseProjection extends GraphQLResponseProjection {

    public StarredRepositoryEdgeResponseProjection() {
    }

    @Override
    public StarredRepositoryEdgeResponseProjection all$() {
        return all$(3);
    }

    @Override
    public StarredRepositoryEdgeResponseProjection all$(int maxDepth) {
        this.cursor();
        if (projectionDepthOnFields.getOrDefault("StarredRepositoryEdgeResponseProjection.RepositoryResponseProjection.node", 0) <= maxDepth) {
            projectionDepthOnFields.put("StarredRepositoryEdgeResponseProjection.RepositoryResponseProjection.node", projectionDepthOnFields.getOrDefault("StarredRepositoryEdgeResponseProjection.RepositoryResponseProjection.node", 0) + 1);
            this.node(new RepositoryResponseProjection().all$(maxDepth - projectionDepthOnFields.getOrDefault("StarredRepositoryEdgeResponseProjection.RepositoryResponseProjection.node", 0)));
        }
        this.starredAt();
        this.typename();
        return this;
    }

    public StarredRepositoryEdgeResponseProjection cursor() {
        return cursor(null);
    }

    public StarredRepositoryEdgeResponseProjection cursor(String alias) {
        fields.add(new GraphQLResponseField("cursor").alias(alias));
        return this;
    }

    public StarredRepositoryEdgeResponseProjection node(RepositoryResponseProjection subProjection) {
        return node(null, subProjection);
    }

    public StarredRepositoryEdgeResponseProjection node(String alias, RepositoryResponseProjection subProjection) {
        fields.add(new GraphQLResponseField("node").alias(alias).projection(subProjection));
        return this;
    }

    public StarredRepositoryEdgeResponseProjection starredAt() {
        return starredAt(null);
    }

    public StarredRepositoryEdgeResponseProjection starredAt(String alias) {
        fields.add(new GraphQLResponseField("starredAt").alias(alias));
        return this;
    }

    public StarredRepositoryEdgeResponseProjection typename() {
        return typename(null);
    }

    public StarredRepositoryEdgeResponseProjection typename(String alias) {
        fields.add(new GraphQLResponseField("__typename").alias(alias));
        return this;
    }


}

File: src/main/kotlin-graphql/io/github/graphql/kotlin/model/StarredRepositoryEdgeResponseProjection.kt
package io.github.graphql.kotlin.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseField
import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLResponseProjection
import java.util.Objects

/**
 * Response projection for StarredRepositoryEdge
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-07T19:48:13+0800"
)
open class StarredRepositoryEdgeResponseProjection : GraphQLResponseProjection() {

    override fun `all$`(): StarredRepositoryEdgeResponseProjection = `all$`(3)

    override fun `all$`(maxDepth: Int): StarredRepositoryEdgeResponseProjection {
        this.cursor()
        if (projectionDepthOnFields.getOrDefault("StarredRepositoryEdgeResponseProjection.RepositoryResponseProjection.node", 0) <= maxDepth) {
            projectionDepthOnFields["StarredRepositoryEdgeResponseProjection.RepositoryResponseProjection.node"] = projectionDepthOnFields.getOrDefault("StarredRepositoryEdgeResponseProjection.RepositoryResponseProjection.node", 0) + 1
            this.node(RepositoryResponseProjection().`all$`(maxDepth - projectionDepthOnFields.getOrDefault("StarredRepositoryEdgeResponseProjection.RepositoryResponseProjection.node", 0)))
        }
        this.starredAt()
        this.typename()
        return this
    }

    fun cursor(): StarredRepositoryEdgeResponseProjection = cursor(null)

    fun cursor(alias: String?): StarredRepositoryEdgeResponseProjection {
        fields.add(GraphQLResponseField("cursor").alias(alias))
        return this
    }

    fun node(subProjection: RepositoryResponseProjection): StarredRepositoryEdgeResponseProjection = node(null, subProjection)

    fun node(alias: String?, subProjection: RepositoryResponseProjection): StarredRepositoryEdgeResponseProjection {
        fields.add(GraphQLResponseField("node").alias(alias).projection(subProjection))
        return this
    }

    fun starredAt(): StarredRepositoryEdgeResponseProjection = starredAt(null)

    fun starredAt(alias: String?): StarredRepositoryEdgeResponseProjection {
        fields.add(GraphQLResponseField("starredAt").alias(alias))
        return this
    }

    fun typename(): StarredRepositoryEdgeResponseProjection = typename(null)

    fun typename(alias: String?): StarredRepositoryEdgeResponseProjection {
        fields.add(GraphQLResponseField("__typename").alias(alias))
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as StarredRepositoryEdgeResponseProjection
        return Objects.equals(fields, that.fields)
    }

    override fun hashCode(): Int = Objects.hash(fields)

}
